pkg/util/operatorcondition.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"context"
+
+	operatorframeworkv2 "github.com/operator-framework/api/pkg/operators/v2"
+	"github.com/operator-framework/operator-lib/conditions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Condition - We just need the Set method in our code.
+type Condition interface {
+	Set(ctx context.Context, status metav1.ConditionStatus, reason, message string) error
+}
+
+// OperatorCondition wraps operator-lib's Condition to make it not crash,
+// when running locally or in Kubernetes without OLM.
+type OperatorCondition struct {
+	cond conditions.Condition
+}
+
+const (
+	UpgradeableCondition = "Upgradeable"
+
+	UpgradeableInitReason  = "Initializing"
+	UpgradeableInitMessage = "The hyperConverged cluster operator is starting up"
+
+	UpgradeableUpgradingReason  = "AlreadyPerformingUpgrade"
+	UpgradeableUpgradingMessage = "upgrading the hyperConverged cluster operator to version "
+
+	UpgradeableAllowReason  = "Upgradeable"
+	UpgradeableAllowMessage = ""
+)
```

Comment:
complex question,
do we also have to think about an upgrade failed reason where after, a long timeout, we mark the upgrade process as failed?
Is the OLM going to do something with that?

Reply:
I'm not sure we know right now. We can add this logic in the future.